src/ADAL.PCL/TokenCache.cs

```diff
@@ -357,17 +357,18 @@ internal void StoreToCache(AuthenticationResultEx result, string authority, stri
             TokenCacheKey tokenCacheKey = new TokenCacheKey(authority, resource, clientId, subjectType, result.Result.UserInfo);
             this.tokenCacheDictionary[tokenCacheKey] = result;
             PlatformPlugin.Logger.Verbose(callState, "An item was stored in the cache");
-            this.UpdateCachedMrrtRefreshTokens(result, authority, clientId, subjectType);
+            this.UpdateCachedMrrtRefreshTokens(result, clientId, subjectType);
 
             this.HasStateChanged = true;
         }
 
-        private void UpdateCachedMrrtRefreshTokens(AuthenticationResultEx result, string authority, string clientId, TokenSubjectType subjectType)
+        private void UpdateCachedMrrtRefreshTokens(AuthenticationResultEx result, string clientId, TokenSubjectType subjectType)
         {
             if (result.Result.UserInfo != null && result.IsMultipleResourceRefreshToken)
             {
+                //pass null for authority to update the token for all the tenants
```

Comment:
There will be multiple entries for the most part and No exception is thrown. RT must be updated for all tenants associated with the client id.